_posts/2021-10-06-#23157.md

```diff
@@ -10,15 +10,80 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
+We recently [looked at PR 22677](./22677), which cuts the circular dependency
+between validation and txmempool. This week's PR was split off from that PR,
+taking a slightly different approach to simplify the `CTxMemPool::check()`
+function.
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- The `CTxMemPool` class is a data structure that stores the set of unconfirmed
+  transactions, along with various methods to update or query that set. As well
+  as the transactions themselves, the mempool stores metadata about the
+  transactions (such as ancestor/descendant dependencies) that makes it highly
+  optimized for various operations, such as constructing blocks.
+
+- The mempool must only contain non-conflicting, valid transactions which
+  could theoretically be included in a future block. There are therefore
```

Comment:
```suggestion
  could theoretically be included in the next block. There are therefore
```

Reply:
(stricter than future block. must be minable immediately)

Reply:
You're right!